Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/task/SubmissionTask.kt

```diff
@@ -125,10 +125,12 @@ class SubmissionTask @Inject constructor(
         }
     }
 
-    private suspend fun performSubmission(): Result {
+    private suspend fun performSubmission(testType: CoronaTest.Type?): Result {
         val availableTests = coronaTestRepository.coronaTests.first()
         Timber.tag(TAG).v("Available tests: %s", availableTests)
-        val coronaTest = availableTests.firstOrNull { it.isSubmissionAllowed }
+        val coronaTest = availableTests
+            .filter { testType == null || it.type == testType }
```

Comment:
In which case can `testType` be null here?

Reply:
when automatic submission is scheduled on the background

Reply:
I see. Maybe add a comment or have a second function `performSubmissionInBackground()` that is called when the `testType` is `null` ?

Reply:
I've added a comment

Reply:
all tests have a `type` if you filter with `testType == null` the filter result will empty, right? and this will lead to 
`?: throw IllegalStateException("No valid test available to authorize submission")`

Reply:
um, I see